### ISSUE TITLE
GetLegendGraphics wrong font rendering on Windows

### ISSUE THREAD

**Issue by cxcandid** [qgis/QGIS#43349]
QGIS Server 3.18.3-1 (Windows Server 2016) shows wrong font rendering in GetLegendGraphics calls if QT_QPA_FONTDIR ist set to "c:\windows\fonts". Without setting QT_QPA_FONTDIR font rendering is ok, but differs between QGIS Desktop and QGIS Server.

Sample Legend rendering in Lizmap Client:
![Clipboard01](https://user-images.githubusercontent.com/58068164/119236030-842af280-bb35-11eb-9607-0ecc5fff8e91.jpg)

**Comment by cxcandid** [comment]
A deeper investigation shows that it's a possible problem with the default font-family selection. Adding "ITEMFONTFAMILY=Arial" returns correct font rendering.

`http://<myserver>/webgis/index.php/lizmap/service/?repository=<myrepos>&project=<myproject>&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetLegendGraphic&LAYER=pipelines&STYLE=default&EXCEPTIONS=application%2Fvnd.ogc.se_inimage&FORMAT=image%2Fpng&TRANSPARENT=TRUE&WIDTH=150&LAYERFONTSIZE=0&ITEMFONTSIZE=9&SYMBOLSPACE=1&ICONLABELSPACE=2&DPI=96&RULELABEL=AUTO&LAYERSPACE=0&LAYERFONTBOLD=FALSE&LAYERTITLE=FALSE&SCALE=1155581.108416699&ITEMFONTFAMILY=Arial`

**Comment by cxcandid** [comment]
Maybe we could have a system variable to define a default font family for legends (i.e. QGIS_LEGEND_FONTFAMILY=Arial), or is there any other workaround?

**Comment by cxcandid** [comment]
Seems like I found the source of my font rendering problem today!

QGIS Server takes the first font it finds in the font directory to render the legend.
In my case I had additional fonts installed in c:\windows\fonts (AcadEref.ttf, AIGDT___.ttf, AMDT_Symbols.ttf, AMGDT___.ttf). I removed these font files so that arial.ttf was the first found, with the result that font rendering behaves as expected now.

IMHO a system variable for defining the default font familiy for legends would be really appreciated (i.e. QGIS_LEGEND_FONTFAMILY=Arial).
